core/src/main/scala/cats/syntax/parallel.scala

```diff
@@ -237,35 +241,69 @@ final class ParallelUnorderedFlatSequenceOps[T[_], M[_], A](private val tmta: T[
     Parallel.parUnorderedFlatSequence(tmta)
 }
 
-final class ParallelApOps[M[_], A](private val ma: M[A]) extends AnyVal {
+final class ParallelApOps[M[_], A](protected val ma: M[A]) extends AnyVal with ParallelApOpsBinCompat0[M, A] {
 
-  def &>[B](mb: M[B])(implicit P: Parallel[M]): M[B] =
+  @deprecated("use a NonEmptyParallel-constrained version instead", "2.8.0")
+  protected def &>[B](mb: M[B])(implicit P: Parallel[M]): M[B] =
     P.parProductR(ma)(mb)
 
-  def <&[B](mb: M[B])(implicit P: Parallel[M]): M[A] =
+  @deprecated("use a NonEmptyParallel-constrained version instead", "2.8.0")
+  protected def <&[B](mb: M[B])(implicit P: Parallel[M]): M[A] =
     P.parProductL(ma)(mb)
 
-  def parProductL[B](mb: M[B])(implicit P: Parallel[M]): M[A] =
+  @deprecated("use a NonEmptyParallel-constrained version instead", "2.8.0")
+  protected def parProductL[B](mb: M[B])(implicit P: Parallel[M]): M[A] =
     P.parProductL(ma)(mb)
 
-  def parProductR[B](mb: M[B])(implicit P: Parallel[M]): M[B] =
+  @deprecated("use a NonEmptyParallel-constrained version instead", "2.8.0")
+  protected def parProductR[B](mb: M[B])(implicit P: Parallel[M]): M[B] =
     P.parProductR(ma)(mb)
 
-  def parProduct[B](mb: M[B])(implicit P: Parallel[M]): M[(A, B)] =
+  @deprecated("use a NonEmptyParallel-constrained version instead", "2.8.0")
+  protected def parProduct[B](mb: M[B])(implicit P: Parallel[M]): M[(A, B)] =
     Parallel.parProduct(ma, mb)
 
   def parReplicateA(n: Int)(implicit P: Parallel[M]): M[List[A]] =
     Parallel.parReplicateA(n, ma)
 }
 
-final class ParallelApplyOps[M[_], A, B](private val mab: M[A => B]) extends AnyVal {
-  def <&>(ma: M[A])(implicit P: Parallel[M]): M[B] =
+sealed private[syntax] trait ParallelApOpsBinCompat0[M[_], A] extends Any { self: ParallelApOps[M, A] =>
```

Comment:
I'm thinking about a thing... Looks like these updated methods are not designed for the `Parallel` typeclass anymore. They all target `NonEmptyParallel` instead. So perhaps it would be better to create a separate extension holder for that, e.g.
```
final class NonEmptyParallelApOps[M[_], A] private[syntax] (private val ma: M[A]) extends AnyVal {
  def &>[B](mb: M[B])(implicit P: NonEmptyParallel[M]): M[B] = /* and so on */  
}
```
And the former class would remain almost the same as before with the corresponding methods hidden and deprecated:
```
final class ParallelApOps[M[_], A](private val ma: M[A]) extends AnyVal {

  @deprecated("use a NonEmptyParallel-constrained version instead", "2.8.0")
  protected def &>[B](mb: M[B])(implicit P: Parallel[M]): M[B] = // ....
```

In that case we wouldn't need such a quite weird `ParallelApOpsBinCompat0` mixin trait anymore and the overall syntax would become a bit clearer.

And the same is applicable for `ParallelApplyOps` below.

So wdyt?

Reply:
sure that makes sense. I'll do that